### PR TITLE
New: add more storagemap get methods

### PR DESCRIPTION
**Pull request by Jim8y**
Closes https://github.com/neo-project/neo-devpack-dotnet/issues/833

### CODE REVIEW

File: tests/Neo.SmartContract.Framework.UnitTests/TestClasses/Contract_Storage.cs [hunk 1]
```diff
@@ -1,7 +1,9 @@
 using Neo.SmartContract.Framework.Services;
+using Neo.SmartContract.Framework;
 
 namespace Neo.SmartContract.Framework.UnitTests.TestClasses
 {
+
     public class Contract_Storage : SmartContract
```
Comment (shargon) [inline tests/Neo.SmartContract.Framework.UnitTests/TestClasses/Contract_Storage.cs:7]: @Liaojinghui  you love this empty lines, isn't it?

```suggestion
    public class Contract_Storage : SmartContract
```

File: src/Neo.SmartContract.Framework/Services/StorageMap.cs [hunk 2]
```diff
@@ -114,6 +114,89 @@ public class StorageMap
...
         public extern ByteString Get(ByteString key);
 
+        [CallingConvention(CallingConvention.Cdecl)]
+        [OpCode(OpCode.UNPACK)]
+        [OpCode(OpCode.DROP)]
+        [OpCode(OpCode.REVERSE3)]
+        [OpCode(OpCode.CAT)]
+        [OpCode(OpCode.SWAP)]
+        [Syscall("System.Storage.Get")]
+        public extern ByteString GetUInt160(ByteString key);
```
Comment (shargon) [inline src/Neo.SmartContract.Framework/Services/StorageMap.cs:124]: It must be UInt160

File: src/Neo.SmartContract.Framework/Services/StorageMap.cs [hunk 3]
```diff
@@ -114,6 +114,89 @@ public class StorageMap
...
+
+        [CallingConvention(CallingConvention.Cdecl)]
+        [OpCode(OpCode.UNPACK)]
+        [OpCode(OpCode.DROP)]
+        [OpCode(OpCode.REVERSE3)]
+        [OpCode(OpCode.CAT)]
+        [OpCode(OpCode.SWAP)]
+        [Syscall("System.Storage.Get")]
+        public extern ByteString GetUInt256(ByteString key);
```
Comment (shargon) [inline src/Neo.SmartContract.Framework/Services/StorageMap.cs:133]: It must be UInt256

File: src/Neo.SmartContract.Framework/Services/StorageMap.cs [hunk 4]
```diff
@@ -114,6 +114,89 @@ public class StorageMap
...
+
+        [CallingConvention(CallingConvention.Cdecl)]
+        [OpCode(OpCode.UNPACK)]
+        [OpCode(OpCode.DROP)]
+        [OpCode(OpCode.REVERSE3)]
+        [OpCode(OpCode.CAT)]
+        [OpCode(OpCode.SWAP)]
+        [Syscall("System.Storage.Get")]
+        public extern ByteString GetECPoint(ByteString key);
```
Comment (shargon) [inline src/Neo.SmartContract.Framework/Services/StorageMap.cs:142]: It must be ECPoint

File: src/Neo.SmartContract.Framework/Services/StorageMap.cs [hunk 5]
```diff
@@ -114,6 +114,89 @@ public class StorageMap
...
+
+        [CallingConvention(CallingConvention.Cdecl)]
+        [OpCode(OpCode.UNPACK)]
+        [OpCode(OpCode.DROP)]
+        [OpCode(OpCode.REVERSE3)]
+        [OpCode(OpCode.CAT)]
+        [OpCode(OpCode.SWAP)]
+        [Syscall("System.Storage.Get")]
+        [OpCode(OpCode.CONVERT, StackItemType.Integer)]
+        public extern int GetInteger(ByteString key);
```
Comment (shargon) [inline src/Neo.SmartContract.Framework/Services/StorageMap.cs:171]:
```suggestion
        public extern int GetInt32(ByteString key);
```

File: src/Neo.SmartContract.Framework/Services/StorageMap.cs [hunk 6]
```diff
@@ -114,6 +114,89 @@ public class StorageMap
...
+        [OpCode(OpCode.SWAP)]
+        [Syscall("System.Storage.Get")]
+        [OpCode(OpCode.DUP)]
+        [OpCode(OpCode.ISNULL)]
+        [OpCode(OpCode.JMPIFNOT, "0x06")]
+        [OpCode(OpCode.DROP)]
+        [OpCode(OpCode.PUSH0)]
+        [OpCode(OpCode.JMP, "0x04")]
+        [OpCode(OpCode.CONVERT, StackItemType.Integer)]
+        public extern int GetIntegerOrZero(ByteString key);
```
Comment (shargon) [inline src/Neo.SmartContract.Framework/Services/StorageMap.cs:198]:
```suggestion
        public extern int GetInt32OrZero(ByteString key);
```

Comment (Liaojinghui) [reply src/Neo.SmartContract.Framework/Services/StorageMap.cs:198]: Maybe we should use Biginteger?

**Review by shargon — APPROVED** [review]

**Comment by vncoelho** [comment]
Closes https://github.com/neo-project/neo-devpack-dotnet/issues/769 as well?

File: src/Neo.SmartContract.Framework/Services/StorageMap.cs [hunk 7]
```diff
@@ -197,6 +364,42 @@ public object GetObject(byte[] key)
...
+        public extern void Put(ByteString key, bool value);
+
+        [CallingConvention(CallingConvention.Cdecl)]
+        [OpCode(OpCode.UNPACK)]
+        [OpCode(OpCode.DROP)]
+        [OpCode(OpCode.REVERSE3)]
+        [OpCode(OpCode.CAT)]
+        [OpCode(OpCode.SWAP)]
+        [Syscall("System.Storage.Put")]
+        public extern void Put(byte[] key, bool value);
```
Comment (vncoelho) [inline src/Neo.SmartContract.Framework/Services/StorageMap.cs:383]: Considering that opcodes were equal, weren't it be better if aggregated? Maybe there is a simple way for such cases.

**Pull request merged**